### PR TITLE
PM-15022 Auto login when user completes a YubiKey login trigger.

### PR DESCRIPTION
**Pull request by dseverns-livefront**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-15022

## 📔 Objective
- After the user either taps the key to the device or presses the button while attached via USB, attempt login.

Required for any UI changes; delete if not applicable. Use fixed width images for better display. -->

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt [hunk 1]
```diff
@@ -254,6 +255,9 @@ private fun TwoFactorLoginScreenContent(
                 modifier = Modifier
                     .padding(horizontal = 16.dp)
                     .fillMaxWidth(),
+                keyboardActions = KeyboardActions(
+                    onDone = { onContinueButtonClick() },
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt:259]: This worked out nicely then?

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt [hunk 2]
```diff
@@ -254,6 +255,9 @@ private fun TwoFactorLoginScreenContent(
                 modifier = Modifier
                     .padding(horizontal = 16.dp)
                     .fillMaxWidth(),
+                keyboardActions = KeyboardActions(
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt:258]: Super minor but can we lift this up so the modifier is last?

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt [hunk 3]
```diff
@@ -367,6 +367,7 @@ class TwoFactorLoginViewModel @Inject constructor(
                 isContinueButtonEnabled = true,
             )
         }
+        initiateLogin()
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginViewModel.kt:370]: 👍

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt (hunk 1 above)
Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt:259]: yes! like a dream.

File: app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt (hunk 2 above)
Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/twofactorlogin/TwoFactorLoginScreen.kt:255]: done!

**Review by david-livefront — APPROVED** [review]

**Pull request merged**